erts/emulator/beam/erl_db_util.c

```diff
@@ -2804,6 +2805,63 @@ Eterm db_prog_match(Process *c_p,
                }
             }
             break;
+        case matchCurrentStacktrace: {
+            Uint sz;
+            Uint heap_size;
+            Eterm mfa;
+            Eterm res;
+            struct StackTrace *s;
+            int depth;
+            FunctionInfo* stk;
+            FunctionInfo* stkp;
+
+            ASSERT(c_p == self);
+
+            depth = unsigned_val(esp[-1]);
```

Comment:
We forgot to consume the depth constant, so the stack was overrun when writing the result.
```suggestion
            depth = unsigned_val(esp[-1]);
            esp--;
```